web/static/js/components/questionnaires/QuestionnaireEditor.jsx

```diff
@@ -77,8 +71,8 @@ class QuestionnaireEditor extends Component {
     // I couldn't find a better way. Ideally this should be a sort of "callback"
     // to the addStep method, without involving additional component state handling
     // or explicit management via Redux reducers.
-    const questionnaireData = newProps.questionnaire.data
-    if (this.state.addingStep && questionnaireData != null && questionnaireData.steps != null && questionnaireData.steps.length > 0) {
+    const questionnaireData = newProps.questionnaire
+    if (this.state.addingStep && questionnaireData && questionnaireData.steps != null && questionnaireData.steps.length > 0) {
```

Comment:
I'll make a regression over this once it's merged and see if there is a breaking case.